docs/source/development/releasing.rst

```diff
@@ -185,6 +185,79 @@ Voting and approval
 Start the vote thread on dev@arrow.apache.org and supply instructions for verifying the integrity of the release.
 Approval requires a net of 3 +1 votes from PMC members. A release cannot be vetoed.
 
+How to Verify Release Candidates
+--------------------------------
+
+#. Install dependencies. If you are not using Conda (see below), you will need to install all dependencies to build and verify all languages. Either way, at minimum, you will need:
+
+   - cURL
+   - Docker (to verify binaries)
+   - Git
+   - GnuPG
+   - shasum (built into macOS) or sha256sum/sha512sum (on Linux)
+
+   If not using Conda, you will need roughly:
+
+   - C and C++ compilers (or the equivalent of ``build-essential`` for your platform)
+   - Python 3
+   - Ruby with headers
+   - bundler, rake, red-arrow, and test-unit Ruby gems
+   - GLib and gobject-introspection with headers
+   - Java JRE and JDK (Java 8+)
+   - Go
+   - CMake, ninja-build, libpq (with headers), SQLite (with headers)
+
+#. Download the source archive from dist.apache.org. A link to the source release is given in the vote email::
+
+     $ wget https://dist.apache.org/repos/dist/dev/arrow/apache-arrow-adbc-0.1.0-rc6/apache-arrow-adbc-0.1.0.tar.gz
```

Comment:
How about using `dev/release/verify-release-candidate.sh` in cloned apache/arrow-adbc instead?

```bash
$ git clone https://github.com/apache/arrow-adbc.git
$ cd arrow-adbc
$ dev/release/verify-release-candidate.sh 0.1.0 6
```

Apache Arrow C++ uses this approach.

With this approach, we can remove most instructions from this documents such as download source archive, verify sign/checksums and extract archive. Because they are done in the verification script.

Reply:
It turns out I've been doing extra work for Arrow this whole time :sweat_smile: I'll remove these steps and update this.